apis/refs/v1beta1/resourceRef.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+type MultiTypeResourceRef struct {
+	// The relative resource name of the referenced resource, when not managed
+	// by Config Connector.
+	// This field must be set when 'name' field is not set.
+	// +optional
+	External string `json:"external,omitempty"`
+	// The apiVersion (<group>/<version>) of the referenced resource.
+	// +required
+	ApiVersion string `json:"apiVersion,omitempty"`
+	// The kind of the referenced resource.
+	// +required
+	Kind string `json:"kind,omitempty"`
```

Comment:
We do not want to support generic Kind. And maybe we don't need this field entirely. See my other comment about using KCC IAM resources

Reply:
Used three different refs to represent this field instead.